### ISSUE TITLE
Music files getting corrupt on default music app

### ISSUE THREAD

**Issue by dkrmdi** [Cimmerian-Iter/Vita-troubleshooting-guide#1]
I still do not know what is triggering it. After leaving vita idle for a long time, I opened music app and played a song. It did not play any song, and displayed error pop up. I think it just said it could not play song.

I reboot my vita and went back to music app. The music I attempted to play before the reboot became corrupt. But I could play other songs.

I wonder what is causing this. This happened twice so far.

**Comment by Cimmerian-Iter** [comment]
do you use a sd2vita?
do you use media importer?

**Comment by rightsider** [comment]
Same thing used to happen for me years ago on ofw and recently on 3.65 enso it happened again. Makes me wonder what is the reason behind that too.

**Comment by Cimmerian-Iter** [comment]
did you import the music using media importer? and do you use a sd2vita?

As far as i know, the bug of the corrupted music that vanish if we reboot is caused by media importer.

And as for the completly corrupted music that doesn't works even after a reboot, try to play it using vitashell. 
it could be caused by the vita itself as there has been cases of corruption (tho there is a way to solve it)

**Comment by rightsider** [comment]
It used to happen to me back in 2013-2014 when I was downloading music directly from browser using Sony MC, everything was working fine for a long time, but then one song randomly got corrupted. Rebooting didn't change a thing. 

About one week ago one of the songs got corrupted again. I'm using sd2vita (yamt driver). I got music from different sources this time (Network Media Player, QCMA, using Vitashell's "export media" feature). Again rebooting didn't really do anything. I didn't even remember what song it was and I didn't feel like playing "what song is missing" game checking every file one by one in Vitashell so I deleted that song a few days later. Honestly, I doubt that actually corrupts the file, might be just some issue with the database.

**Comment by Cimmerian-Iter** [comment]
https://www.reddit.com/r/vitahacks/comments/ezr7xf/psa_check_your_sd_cardmemory_stick_often_since/?utm_source=share&utm_medium=ios_app&utm_name=iossmf

there is some record about corruption case. It looks like it is a vita problem. Difficulties to handle fat function and can corrupt memorycard.

Well fortunately with the sd2vita it will be easier to fix this problem.

**Comment by rightsider** [comment]
I have been using vita on daily basis since 2012 and music was the only thing to ever get corrupted. Everything else started being corupted when I started transfering stuff  using vitashell to manage files. I think the way vita handles files using QCMA and not letting user commit any changes directly decreases the probability of getting corruptions. CHKDSK is very useful and helps to solve many problems with filesystem ...it might be able to fix this problem. I'll try to see if it does next time. Though, CHKDSK can also corrupt files itself while fixing (each time it damaged something else everytime I run it (repatch files/savedata), but apparently even when it does damage them it's still because there was some problem with those sectors before), so I think it's worth to backup important files before using it.

**Comment by Cimmerian-Iter** [comment]
yeah, fortunately i haven't encountered these issues, and it's "a common problem," but only affect 20% of the userbase.
That's the reason too begind the creation of content manager and it's weird sort process. 
Everything is run directly by the psvita and it manage all the I/O interaction since cma on pc is just a client to make the link.

**Comment by dkrmdi** [comment]
I use sd2vita, but I use official CMA when I am transfering files. I care about making a playlist and album arts.

I am on 3.60 enso.

**Comment by dkrmdi** [comment]
I should also mention that the music files were in fact not corrupted. The music app only declared that they were corrupt.

I copied the "corrupt" files back to my pc, and I was able to play the files without problem.

It's probably some directory issue.

**Comment by Cimmerian-Iter** [comment]
Yeah I know they aren't corrupted. It might be the database that mess up itself.

Try to rebuild database from the safe mode and if it doesn't work then try the manipulation in this PSA to clean your sdcard (won't format )
https://www.reddit.com/r/vitahacks/comments/ezr7xf/psa_check_your_sd_cardmemory_stick_often_since/?utm_source=share&utm_medium=ios_app&utm_name=iossmf
